File: JavaScript/4.js
/*
Task Four: Largest Palindrome Product

A palindromic number reads the same both ways. The largest palindrome
made from the product of two 2-digit numbers is 9009 = 91 * 99.

Find the largest palindrome made from the product of two 3-digit numbers.
*/

const isPalindrome = (number) => {
	const string = String(number)
	return string == string.split('').reverse().join('');
};

const palindromes = []

for (let outer = 100; outer < 999; outer++) {
	for (let inner = 100; inner < 999; inner++) {

		const value = outer * inner;
		if (isPalindrome(value)) palindromes.push(value);
	};
};

const sorted = palindromes.sort((a, b) => b - a);
console.log( sorted[0] );

File: Python/4.py
"""
Task Four: Largest Palindrome Product

A palindromic number reads the same both ways. The largest palindrome
made from the product of two 2-digit numbers is 9009 = 91 * 99.

Find the largest palindrome made from the product of two 3-digit numbers.
"""

def isPalindrome(string):
	return string == string[::-1]

palindromes = []

for outer in range(100, 999):
	for inner in range(100, 999):

		if (isPalindrome( str(outer * inner) )):
			palindromes.append(outer * inner)

print( max(palindromes) )
